audformat/core/utils.py

```diff
@@ -1255,21 +1250,38 @@ def union(
                     ('f4', '0 days 00:00:00', '0 days 00:00:01'),
                     ('f3', '0 days 00:00:01', '0 days 00:00:02')],
                    names=['file', 'start', 'end'])
+        >>> union(
+        ...     [
+        ...         pd.Index(['spk1'], name='speaker'),
+        ...         pd.MultiIndex.from_arrays([['spk2']], names=['speaker']),
+        ...     ]
+        ... )
+        Index(['spk1', 'spk2'], dtype='object', name='speaker')
 
     """
     if not objs:
         return filewise_index()
 
-    types = [index_type(obj) for obj in objs]
+    objs = [obj if isinstance(obj, pd.Index) else obj.index for obj in objs]
+
+    if len(objs) == 1:
+        return objs[0]
 
-    if len(set(types)) != 1:
-        objs = [to_segmented_index(obj) for obj in objs]
+    # Convert filewise index to segmented
+    # if a segmented index is present
+    if any([is_segmented_index(obj) for obj in objs]):
+        for n, obj in enumerate(objs):
+            if is_filewise_index(obj):
+                objs[n] = to_segmented_index(obj)
 
     # Combine all MultiIndex entries and drop duplicates afterwards,
     # faster than using index.union(),
     # compare https://github.com/audeering/audformat/pull/98
     df = pd.concat([o.to_frame() for o in objs])
-    index = df.index
+    if len(df.columns) > 1:
+        index = pd.MultiIndex.from_frame(df)
```

Comment:
It might be that this will slow down combining only segmented indices, as before we could simply use the index returned by `pd.concat()`. So maybe we are faster if we add another check if all indices are segmented and use `index = df.index` in that case.

Reply:
I guess, I should also do some execution time testing for this.